servicediscovery/mdns/mdns.go

```diff
@@ -0,0 +1,75 @@
+// ------------------------------------------------------------
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+// ------------------------------------------------------------
+
+package mdns
+
+import (
+	"context"
+	"fmt"
+	"sync"
+	"time"
+
+	"github.com/dapr/components-contrib/servicediscovery"
+	"github.com/grandcat/zeroconf"
+)
+
+func NewMDNSResolver() *Resolver {
+	return &Resolver{}
+}
+
+type Resolver struct {
+	Once     sync.Once
+	Resolver *zeroconf.Resolver
```

Comment:
We do not need to expose these members to the external package. I would make these member variable names lower-case.